app/tests/spec/lib/crypto/recovery-keys.js

```diff
@@ -0,0 +1,146 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const {assert} = require('chai');
+const RecoveryKeys = require('lib/crypto/recovery-keys');
+
+const uid = 'aaaaabbbbbcccccdddddeeeeefffff00';
+const recoveryKey = '0123456789ABCDEFGHJKMNPQRSTVWXYZ';
+const recoveryData = {
+  kB: '000000111111222222333333444444555555666666777777888888999999ABCD'
+};
+const expectedRecoveryKeyId = 'c3cdf70fbf0789046156ac68fc3db8fa';
+const expectedBundlePart2 = '7t3Mu6qZiHdmVUQz';
+const expectedBundlePart3 = 's4M8182Y1Oebar7jtlyLB1twRcDsi0jNnfSTVNBZUzDbHm_LUPUa3dvdUoSHBeBn6-6FE2-exibwoRjuErIj7W-DGBjmbz6CzA';
+const encryptionOptions = {
+  iv: 'eeddccbbaa99887766554433'
+};
+
+describe('lib/crypto/recovery-keys', () => {
```

Comment:
I suggest adding a "round trip" sanity check test:
```
    generateRecoveryKey() --> getRecoveryJwk() --> bundleRecoveryData() --> unbundleRecoveryData()
```

I _think_ I see a gap, but I'm not entirely sure ... and the above test would verify it.